functions/issue-scraper.ts

```diff
@@ -0,0 +1,368 @@
+    repositoryName: string;
+    repositoryId: number;
+    assignees: string[];
+    createdAt: string;
+    closedAt: string | null;
+    updatedAt: string;
+  };
+  action: string;
+  sender: {
+    login: string;
+  };
+  repository: {
+    id: number;
+    node_id: string;
+    name: string;
+    full_name: string;
+    owner: {
+      login: string;
+      id: number;
+      type: string;
+      site_admin: boolean;
+    };
+  };
+}
+
+interface IssueNode {
+  id: string;
+  number: number;
+  title: string;
+  body: string;
+  state: string;
+  stateReason: string | null;
+  createdAt: string;
+  updatedAt: string;
+  closedAt: string | null;
+  author: {
+    login: string;
+  } | null;
+  assignees: {
+    nodes: Array<{
+      login: string;
+    }>;
+  };
+  repository: {
+    id: string;
+    name: string;
+    owner: {
+      login: string;
+    };
+  };
+}
+
+interface GraphQlSearchResponse {
+  search: {
+    pageInfo: {
+      hasNextPage: boolean;
+      endCursor: string | null;
+    };
+    nodes: Array<IssueNode>;
+  };
+}
+
+export const corsHeaders = {
+  "Access-Control-Allow-Origin": "*",
+  "Access-Control-Allow-Methods": "GET",
+  "Access-Control-Allow-Headers": "Content-Type",
+};
+
+export async function onRequest(ctx: Context): Promise<Response> {
+  const { request, env } = ctx;
+  try {
+    switch (request.method) {
+      case "POST": {
+        const result = await validatePOST(request);
+        if (!result.isValid || !result.gitHubUser) {
+          return new Response("Unauthorized", {
+            headers: corsHeaders,
+            status: 400,
+          });
+        }
+        const githubUserName = result.gitHubUser.login;
+        const timestamp = result.timestamp; // Unix timestamp in milliseconds
+
+        try {
+          const supabase = new SupabaseClient(env.SUPABASE_URL, env.SUPABASE_KEY);
+          const response = await issueScraper(githubUserName, supabase, env.VOYAGEAI_API_KEY, result.authToken, timestamp);
+          return new Response(response, {
+            headers: corsHeaders,
+            status: 200,
+          });
+        } catch (error) {
+          console.error("Error processing request:", error);
+          return new Response("Internal Server Error", {
+            headers: corsHeaders,
+            status: 500,
+          });
+        }
+      }
+
+      default:
+        return new Response("Method Not Allowed", {
+          headers: corsHeaders,
+          status: 405,
+        });
+    }
+  } catch (error) {
+    console.error("Error processing request:", error);
+    return new Response("Internal Server Error", {
+      headers: corsHeaders,
+      status: 500,
+    });
+  }
+}
+
+function markdownToPlainText(markdown: string | null): string | null {
+  if (!markdown) return markdown;
+  const md = markdownit() as MarkdownItWithPlainText;
+  md.use(plainTextPlugin);
+  md.render(markdown);
+  return md.plainText;
+}
+
+const SEARCH_ISSUES_QUERY = `
```

Comment:
You can add `/* GraphQL */` to help IDEs parse and format GraphQL queries.
```suggestion
const SEARCH_ISSUES_QUERY = /* GraphQL */`
```